docker-compose.yml

```diff
@@ -0,0 +1,150 @@
+      - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=DOCKER:PLAINTEXT,LOCALHOST:PLAINTEXT
+      - KAFKA_INTER_BROKER_LISTENER_NAME=DOCKER
+      - KAFKA_BROKER_ID=1
+      - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+      - KAFKA_AUTO_CREATE_TOPICS_ENABLE=true
+      - KAFKA_ZOOKEEPER_CONNECT=zookeeper:32181
+    ports:
+      - 9092:9092
+      - 29092:29092
+    depends_on:
+      - zookeeper
+
+  init-kafka:
+    profiles:
+      - "kafka"
+    image: confluentinc/cp-kafka
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      kafka-topics --bootstrap-server kafka:29092 --list
+
+      echo -e 'Creating kafka topics'
+      kafka-topics --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.sources.status --replication-factor 1 --partitions 1
+      kafka-topics --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.provisioning.internal.availability-check --replication-factor 1 --partitions 1
+
+      echo -e 'Successfully created the following topics:'
+      kafka-topics --bootstrap-server kafka:29092 --list
+      "
+    depends_on:
+      - kafka
+      - zookeeper    
+  backend:
+    env_file:
+      - config/api.env
+    depends_on:
+      migrate:
+        condition: service_completed_successfully
+      provisioning-db:
+        condition: service_healthy
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    ports:
+      - 8000:8000
+    volumes:
+      - go-modules:/go/pkg/mod # Put modules cache into a separate volume
+
+  migrate: # This service is used to run migrations
+    profiles:
+      - "migrate"
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    env_file:
+      - config/api.env
+    depends_on:
+      provisioning-db:
+        condition: service_healthy
+    entrypoint: [/pbmigrate]
+    volumes:
+      - go-modules:/go/pkg/mod # Put modules cache into a separate volume
+  frontend:
+    environment:
+      - PROV_API_HOST=backend
+    build:
+      context: ../provisioning-frontend
+    ports: 
+      - 1337:1337
+    depends_on:
+    - backend
+
+  sources:
+    profiles:
+      - "sources"
+    build:
+      context: ../sources-api-go
+    ports: 
+      - 8131:8131
+    env_file:
+      - scripts/sources.local.conf
+    depends_on:
+      redis:
+        condition: service_healthy
+      sources-db:
+        condition: service_healthy
+  sources-db:
+    restart: on-failure
+    profiles:
+      - "sources"
+    image: docker.io/postgres:10.5
+    healthcheck:
+      test: [ "CMD", "pg_isready", "-U", "postgres", "-d", "postgres" ]
+      interval: 2s
+      timeout: 2s
+      retries: 10
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=sources_devel
+    volumes:
+      - ./postgres-data-sources:/var/lib/postgresql/data
+    expose:
+      - 5432
+
+volumes:
+  go-modules: # This volume is used to cache go modules
```

Comment:
Where does this volume go? Can we have it in the working directory and `.gitignore` just like postgres? I would very much like to have "everything in one place" so I can clean it up if needed.

Reply:
let's add `go-modules`  caching later on :) I removed it for now